test/utils

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+set -u
+
+cd "$( dirname "${BASH_SOURCE[0]}" )" || return
+
+function teardown() {
+    EXIT_CODE="$(tr -d 0 < "$OUTPUT_DIR/EXIT_CODE.txt")"
+    rm -r "$INPUT_DIR" "$OUTPUT_DIR"
+    exit "${EXIT_CODE:-0}"
+}
+
+INPUT_DIR="$(mktemp -d)"
+OUTPUT_DIR="$(mktemp -d)"
+trap "teardown" EXIT
+
+function run_test() {
+
+    test_name="$1"
+    command_name="$2"
+    shift 2
+
+    echo "⋯ $command_name $test_name"
+
+    (
+        bash ../bin/"$command_name" "$@"
+    ) 1> "$OUTPUT_DIR/$command_name-$test_name.stdout.txt" 2> "$OUTPUT_DIR/$command_name-$test_name.stderr.txt"
+
+    diff -u "fixtures/$command_name-$test_name.stdout.txt" "$OUTPUT_DIR/$command_name-$test_name.stdout.txt"
+    exit_stdout="$?"
+
+    diff -u "fixtures/$command_name-$test_name.stderr.txt" "$OUTPUT_DIR/$command_name-$test_name.stderr.txt"
+    exit_stderr="$?"
+
+    if [ $exit_stdout -eq 0 ] && [ $exit_stderr -eq 0 ] ; then
+        echo "✓ $command_name $test_name"
+        printf %s "0" >> "$OUTPUT_DIR/EXIT_CODE.txt"
+    else
+        echo "✗ $command_name $test_name" >&2
+        printf %s "1" >> "$OUTPUT_DIR/EXIT_CODE.txt"
```

Comment:
```suggestion
        echo "✓ $command_name $test_name"
        echo -n "0" >> "$OUTPUT_DIR/EXIT_CODE.txt"
    else
        echo "✗ $command_name $test_name" >&2
        echo -n "1" >> "$OUTPUT_DIR/EXIT_CODE.txt"
```

Reply:
Not sure why you would want to do that:

https://unix.stackexchange.com/questions/65803/why-is-printf-better-than-echo